### PR TITLE
flatbuffers: 1.12.0 -> 2.0.0

### PR DESCRIPTION
**Pull request by cpcloud**
###### Motivation for this change

This package bumps flatbuffers from 1.12.0 -> 2.0.0, and adds backwards compatibility aliases.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 125677` at 3277cf1b run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages marked as broken and skipped:</summary>
  <ul>
    <li>python38Packages.spacy-transformers</li>
    <li>python38Packages.tensorflowWithCuda</li>
    <li>python39Packages.tensorflowWithCuda</li>
  </ul>
</details>
<details>
  <summary>55 packages skipped due to time constraints:</summary>
  <ul>
    <li>arrow-cpp</li>
    <li>kodi (kodiPackages.kodi)</li>
    <li>kodi-gbm</li>
    <li>kodi-wayland</li>
    <li>kodiPackages.inputstream-adaptive</li>
    <li>kodiPackages.inputstream-ffmpegdirect</li>
    <li>kodiPackages.inputstream-rtmp</li>
    <li>kodiPackages.joystick</li>
    <li>kodiPackages.pvr-hdhomerun</li>
    <li>kodiPackages.pvr-hts</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>27 packages built successfully:</summary>
  <ul>
    <li>buildbot (python38Packages.buildbot)</li>
    <li>buildbot-full (python38Packages.buildbot-full)</li>
    <li>buildbot-ui (python38Packages.buildbot-ui)</li>
    <li>flatbuffers (flatbuffers_2_0)</li>
    <li>flatbuffers_1_12</li>
    <li>gnome-keysign</li>
    <li>grab-site</li>
    <li>magic-wormhole (python38Packages.magic-wormhole)</li>
    <li>python38Packages.autobahn</li>
    <li>python38Packages.channels</li>
    <li>python38Packages.channels-redis</li>
    <li>python38Packages.daphne</li>
    <li>python38Packages.flatbuffers</li>
    <li>python38Packages.labgrid</li>
    <li>python38Packages.magic-wormhole-mailbox-server</li>
    <li>python39Packages.autobahn</li>
    <li>python39Packages.buildbot</li>
    <li>python39Packages.buildbot-full</li>
    <li>python39Packages.buildbot-ui</li>
    <li>python39Packages.channels</li>
    <li>python39Packages.channels-redis</li>
    <li>python39Packages.daphne</li>
    <li>python39Packages.flatbuffers</li>
    <li>python39Packages.labgrid</li>
    <li>python39Packages.magic-wormhole</li>
    <li>python39Packages.magic-wormhole-mailbox-server</li>
    <li>vowpal-wabbit</li>
  </ul>
</details>

File: pkgs/development/libraries/flatbuffers/1.12.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    })
+    (fetchpatch {
+      # Fixed a compilation error with GCC 10.0 to 11.0. July 6, 2020.
+      # Should be included in the next release after 1.12.0
+      url = "https://github.com/google/flatbuffers/pull/6020/commits/44c7a4cf439b0a298720b5a448bcc243a882b0c9.patch";
+      sha256 = "126xwkvnlc4ignjhxv9jygfd9j6kr1jx39hyk0ddpcmvzfqsccf4";
+    })
+  ];
+
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/flatbuffers/1.12.nix:23]: Why are tests disabled?

File: pkgs/development/libraries/flatbuffers/generic.nix [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  inherit version;
+
+  src = fetchFromGitHub {
+    owner = "google";
+    repo = "flatbuffers";
+    rev = "v${version}";
+    inherit sha256;
+  };
+
+  preConfigure = lib.optional stdenv.buildPlatform.isDarwin ''
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/flatbuffers/generic.nix:22]: Can we remove the file uncondional?

File: pkgs/development/libraries/flatbuffers/1.12.nix (hunk 1 above)
Comment (cpcloud) [reply pkgs/development/libraries/flatbuffers/1.12.nix:23]: It seems like they didn't build in the past. I preserved the existing behavior for 1.12.0.

File: pkgs/development/libraries/flatbuffers/generic.nix (hunk 2 above)
Comment (cpcloud) [reply pkgs/development/libraries/flatbuffers/generic.nix:22]: Probably. `BUILD` is almost certainly a bazel file.

Comment (cpcloud) [reply pkgs/development/libraries/flatbuffers/generic.nix]: It seems like this was renamed to `BUILD.bazel`.

Comment (cpcloud) [reply pkgs/development/libraries/flatbuffers/generic.nix]: I removed the `preConfigure` altogether, let's see what happens.

**Comment by risicle** [comment]
On macos 10.15:

```
builder for '/nix/store/02z3ashvrxwy3h78m4d6ihqkilmmcn5d-flatbuffers-1.12.0.drv' failed with exit code 1; last 10 log lines:
  patching file tests/test.cpp
  Hunk #1 succeeded at 3205 (offset -17 lines).
  patching file tests/union_vector/union_vector_generated.h
  applying patch /nix/store/m6rm5vc10a2qjkwjy03l684ack1s5691-44c7a4cf439b0a298720b5a448bcc243a882b0c9.patch
  patching file include/flatbuffers/base.h
  Hunk #1 succeeded at 375 (offset -6 lines).
  Hunk #2 succeeded at 394 (offset -6 lines).
  configuring
  fixing cmake files...
  mkdir: cannot create directory 'build': File exists
```

**Comment by risicle** [comment]
(no other new build failures on macos 10.15)

**Comment by cpcloud** [comment]
@risicle Should that cause a failed CI? It seems like it should.

**Comment by cpcloud** [comment]
I guess another question is: is the file system on macos case insensitive, so the conflicting file is actually `BUILD`?

**Comment by risicle** [comment]
:grin: now 2.0.0 fails with

```
rm: cannot remove 'BUILD': No such file or directory
```

File: pkgs/development/libraries/flatbuffers/generic.nix (hunk 2 above)
Comment (cpcloud) [reply pkgs/development/libraries/flatbuffers/generic.nix]: We can't remove this, see the top level discussion.

**Comment by risicle** [comment]
`nixpkgs-review` happy on macos 10.15 now

**Pull request merged**